crates/webidl/src/util.rs

```diff
@@ -792,3 +579,39 @@ fn flag_slices_immutable(ty: &mut IdlType) {
         _ => {}
     }
 }
+
+pub fn required_doc_string(features: &BTreeSet<String>) -> Option<String> {
+    if features.len() == 0 {
+        return None;
+    }
+    let list = features
+        .iter()
+        .map(|ident| format!("`{}`", ident))
+        .collect::<Vec<_>>()
+        .join(", ");
+    Some(format!(
+        "\n\n*This API requires the following crate features \
+         to be activated: {}*",
+        list,
+    ))
+}
+
+pub fn get_cfg_features(features: &BTreeSet<String>) -> Option<syn::Attribute> {
+    let len = features.len();
+
+    if len == 0 {
+        None
+
+    } else {
+        let features = features.into_iter()
+            .map(|feature| quote!( feature = #feature, ))
+            .collect::<TokenStream>();
+
+        if len == 1 {
+            Some(syn::parse_quote!( #[cfg(#features)] ))
+
+        } else {
+            Some(syn::parse_quote!( #[cfg(all(#features))] ))
```

Comment:
Technically these are both the same thing, so it should be fine to avoid the `if len == 1` condition here.

Reply:
Yes, but I tried hard to generate idiomatic code, similar to the sort of code you would manually write.

Reply:
Ah ok, mind leaving a comment explaining such?